lib/gateways/snapshot-gateway.js

```diff
@@ -60,16 +60,18 @@ class SnapshotGateway {
 
     const { Items } = await this.client.query(request).promise();
 
-    const snapshots = await Promise.all(
-      Items.filter(s => {
-        if (!s.systemIds || s.systemIds.length === 0) {
-          return true;
-        }
-        return s.systemIds.some(id => systemIds.includes(id));
-      })
-        .map(s => createSnapshotFromModel(s))
-        .sort((s1, s2) => (s1.created < s2.created ? 1 : -1))
-    );
+    const matchBySystemId = s => {
+      if (!s.systemIds || s.systemIds.length === 0) return true;
+      return s.systemIds.some(id => systemIds.includes(id));
+    };
+    const matches = await Promise.all(Items.filter(matchBySystemId));
+
+    // because we're using a GSI to find, we need to then do a get
+    // on each match to get the full data model back
+    const snapshots = (
+      await Promise.all(matches.map(m => this.get(m)))
+    ).sort((s1, s2) => (s1.created < s2.created ? 1 : -1));
```

Comment:
This could become a bottleneck if a customer ends up with a large number of snapshots, but probably fine for 99% of cases.